### PR TITLE
fix: offline path should be relative to current path not root

### PR DESCRIPTION
**Pull request by haijian-vaadin**
Fixes https://github.com/vaadin/flow/issues/10177

### CODE REVIEW

File: flow-server/src/main/resources/com/vaadin/flow/server/frontend/sw.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ const rewriteBaseHref = async (response: Response) => {
 };
 
 const appShellPath = '.';
-const offlinePath = OFFLINE_PATH_ENABLED ? '/' + OFFLINE_PATH : appShellPath;
+const offlinePath = OFFLINE_PATH_ENABLED ? OFFLINE_PATH : appShellPath;
```
Comment (knoobie) [inline flow-server/src/main/resources/com/vaadin/flow/server/frontend/sw.ts:44]: What would happen if you open the app for the first time with a nested route like `/my-app/nested/uuid`? Would that register the `sw.ts` on `/my-app` or on `/my-app/nested`? Is there some kind of way to generate this file on start up / change it with the value from e.g. `server.servlet.context-path` to always have the correct value?

Comment (haijian-vaadin) [reply flow-server/src/main/resources/com/vaadin/flow/server/frontend/sw.ts:44]: Thanks for your comment @knoobie. The change shouldn't affect the sw registration, it's just resolution of the path to the custom offline page.

Comment (knoobie) [reply flow-server/src/main/resources/com/vaadin/flow/server/frontend/sw.ts:44]: @haijian-vaadin Thanks for the clarification!

**Comment by Haprog** [comment]
Build failed with:
> Compilation failure
/opt/agent/work/a69251da2b2bb9e9/flow-tests/test-pwa/src/test/java/com/vaadin/flow/pwatest/ui/OfflineWithCustomContextIT.java:[41,8] class PwaTestIT is public, should be declared in a file named PwaTestIT.java

Apparently you forgot to rename class `PwaTestIT` to `OfflineWithCustomContextIT` while renaming the file.

**Comment by haijian-vaadin** [comment]
> Build failed with: `Compilation failure /opt/agent/work/a69251da2b2bb9e9/flow-tests/test-pwa/src/test/java/com/vaadin/flow/pwatest/ui/OfflineWithCustomContextIT.java:[41,8] class PwaTestIT is public, should be declared in a file named PwaTestIT.java`
> 
> Apparently you forgot to rename class `PwaTestIT` to `OfflineWithCustomContextIT` while renaming the file.

Yep, saw it just now, the autosave is not enabled on my new laptop. Let's how it goes with this build.

**Review by Haprog — APPROVED** [review]

**Review by platosha — APPROVED** [review]

**Pull request merged**